server/util.go

```diff
@@ -197,9 +199,39 @@ func CreateSDKLogger(ctx *Context, out io.Writer) (log.Logger, error) {
 	// Check if the CometBFT flag for trace logging is set and enable stack traces if so.
 	opts = append(opts, log.TraceOption(ctx.Viper.GetBool("trace"))) // cmtcli.TraceFlag
 
+	// Error fields should be set under error object
+	zerolog.ErrorFieldName = "error"
+
+	// Add the kind and message field
+	zerolog.ErrorMarshalFunc = func(err error) interface{} {
```

Comment:
We should guard this behind a flag to allow people to opt out of the custom marshalling function in case they aren't using datadog.